docs/source/advanced/java-rxjava.mdx

```diff
@@ -0,0 +1,214 @@
+---
+title: 'Java and RxJava support'
+---
+
+import {MultiCodeBlock} from 'gatsby-theme-apollo-docs';
+
+
+Because Apollo Android is written in Kotlin, it can be used in Java projects. This guide will show you how to do that.
+
+## Use the Java codegen
+
+Apollo Android generates Kotlin code by default, but you can configure it to use the Java codegen instead:
+
+```kotlin:title=build.gradle[.kts]
+apollo {
+  generateKotlinModels.set(false)
+}
+```
+
+## Use RxJava adapters
+
+Apollo Android has a coroutines / Flow based API which isn't well adapted to be used with Java. Or maybe you use RxJava in your Kotlin project. In both cases, you may use Apollo's RxJava adapters.
+
+To do so, add the `apollo-rx2-support` or `apollo-rx3-support` dependency to your project:
+
+```kotlin:title=build.gradle[.kts]
+dependencies {
+  // ...
+
+  // For RxJava 2
+  implementation("com.apollographql.apollo3:apollo-rx2-support:$version")
+
+  // For RxJava 3
+  implementation("com.apollographql.apollo3:apollo-rx3-support:$version")
+}
+```
+
+### Queries, Mutations, Subscriptions
+
+Use the `rxSingle` / `rxFlowable` extensions to execute your operations and get RxJava observables:
+
+<MultiCodeBlock>
+
+```java
+// A query
+ApolloQueryCall<MyQuery.Data> queryCall = client.query(new MyQuery());
+Single<ApolloResponse<MyQuery.Data>> queryResponse = rxSingle(queryCall);
+queryResponse.subscribe( /* ... */ );
+
+// A mutation
+ApolloMutationCall<MyMutation.Data> mutationCall = client.mutate(new MyMutation("my-parameter"));
+Single<ApolloResponse<MyMutation.Data>> mutationResponse = rxSingle(mutationCall);
+mutationResponse.subscribe( /* ... */ );
+
+// A subscription
+ApolloSubscriptionCall<MySubscription.Data> subscriptionCall = client.subscribe(new MySubscription());
+Flowable<ApolloResponse<MySubscription.Data>> subscriptionResponse = rxFlowable(subscriptionCall);
+subscriptionResponse.subscribe( /* ... */ );
+```
+
+```kotlin
+// A query
+val queryCall: ApolloQueryCall<MyQuery.Data> = client.query(MyQuery())
+val queryResponse: Single<ApolloResponse<MyQuery.Data>> = queryCall.rxSingle()
+queryResponse.subscribe(/* ... */)
+
+// A mutation
+val mutationCall = client.mutate(MyMutation("my-parameter"))
+val mutationResponse: Single<ApolloResponse<MyMutation.Data>> = mutationCall.rxSingle()
+mutationResponse.subscribe(/* ... */)
+
+// A subscription
+val subscriptionCall = client.subscribe(MySubscription())
+val subscriptionResponse: Flowable<ApolloResponse<MySubscription.Data>> = subscriptionCall.rxFlowable()
+subscriptionResponse.subscribe(/* ... */)
+```
+</MultiCodeBlock>
+
+### ApolloInterceptor
```

Comment:
For this one an the ones below, I'd mark them as `Experimental` for now as the ergonomics could certainly be improved and we've had limited feedback about them so keeping the ability to change them would be nice.

Reply:
👍  I will add a sentence here.

OK for me to also create/add the annotation on this PR, or preferable to make another PR for that?

Reply:
> OK for me to also create/add the annotation on this PR, or preferable to make another PR for that?
I don't know 😅 . This PR is "only" 215 lines so I guess it's ok to beef it up?